### PR TITLE
Support Enum.map_every/3 and Stream.map_every/3

### PR DESCRIPTION
**Pull request by ottolin**
This pull request adds map_every/3 to Enum and Stream module, fixex #5138

### CODE REVIEW

File: lib/elixir/lib/stream/reducers.ex [hunk 1]
```diff
@@ -120,6 +120,18 @@ defmodule Stream.Reducers do
...
   end
 
+  defmacro map_every(nth, mapper, f \\ nil) do
+    quote do
+      fn
+        (entry, acc(h, n, t)) when n === :first
+                              when n === unquote(nth) ->
+          next_with_acc(unquote(f), unquote(mapper).(entry), h, 1, t)
+        (entry, acc(h, n, t)) ->
+          skip(acc(h, n+1, t))
```
Comment (josevalim) [inline lib/elixir/lib/stream/reducers.ex]: Thank you, however, we should not skip this value and just emit it as is. :)

File: lib/elixir/test/elixir/enum_test.exs [hunk 2]
```diff
@@ -268,6 +268,21 @@ defmodule EnumTest do
     assert Enum.map([1, 2, 3], fn x -> x * 2 end) == [2, 4, 6]
   end
 
+  test "map_every/2" do
+    assert Enum.map_every([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], 2, fn x -> x * 2 end) == [2, 6, 10, 14, 18]
+    assert Enum.map_every([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], 3, fn x -> x * 2 end) == [2, 8, 14, 20]
```
Comment (josevalim) [inline lib/elixir/test/elixir/enum_test.exs]: This should return: [2, 2, 3, 8, 5, 6, 14, 8, 9, 20].

**Comment by josevalim** [comment]
Thank you! I have added one comment related to the implementation!

**Comment by ottolin** [comment]
Oops, I misunderstood the requirement.
Let me modify it.

File: lib/elixir/lib/stream/reducers.ex (hunk 1 above)
Comment (josevalim) [inline lib/elixir/lib/stream/reducers.ex]: I believe this would be the correct expression:

    next_with_acc(unquote(f), entry, h, n+1, t)

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**